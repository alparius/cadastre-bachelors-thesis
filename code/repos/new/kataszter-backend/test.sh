#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
