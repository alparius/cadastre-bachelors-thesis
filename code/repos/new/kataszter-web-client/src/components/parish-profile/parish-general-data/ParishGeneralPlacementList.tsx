import React from "react";
import { NavLink } from "react-router-dom";
import { Container, Grid, Header } from "semantic-ui-react";

import IParishPlacement from "../../../data/parish/ParishPlacement";
import BoldFM from "../../../util/styledMessages/BoldFM";
import ItalicFM from "../../../util/styledMessages/ItalicFM";
import DateDisplay from "../../DateDisplay";

interface IParishGeneralPlacementListProps {
    placements: IParishPlacement[];
    headerKey: string;
}

const ParishGeneralPlacementList: React.FC<IParishGeneralPlacementListProps> = (props: IParishGeneralPlacementListProps) => {
    const { placements, headerKey } = props;

    return (
        <>
            <Header as="h2">
                <ItalicFM id={headerKey} />
            </Header>

            <Container style={{ marginBottom: "2em" }}>
                <Grid>
                    {placements ? (
                        <>
                            <Grid.Row columns={4} style={{ marginBottom: "-1em" }}>
                                <Grid.Column>
                                    <BoldFM id="label.name" />
                                </Grid.Column>
                                <Grid.Column>
                                    <BoldFM id="profile.workInterval" />
                                </Grid.Column>
                                <Grid.Column>
                                    {headerKey !== "label.priest.assistantPlacements" && <BoldFM id="profile.numberOfPeople" />}
                                </Grid.Column>
                            </Grid.Row>
                            {placements.map((placement: IParishPlacement, index: number) => (
                                <Grid.Row columns={4} style={{ marginBottom: "-1em" }} key={placement.priestID + index}>
                                    <Grid.Column>
                                        {placement.priestName ? (
                                            <NavLink to={`/priests/profile/${placement.priestID}`} className="hoverLink">
                                                {placement.priestName}
                                            </NavLink>
                                        ) : (
                                            <ItalicFM id="general.noData" />
                                        )}
                                    </Grid.Column>
                                    <Grid.Column>
                                        <DateDisplay date={placement.startDate} /> - <DateDisplay date={placement.endDate} />
                                    </Grid.Column>
                                    <Grid.Column>{placement.numberOfPeople && <i>{placement.numberOfPeople}</i>}</Grid.Column>
                                    <Grid.Column />
                                </Grid.Row>
                            ))}
                        </>
                    ) : (
                        <ItalicFM id="general.noData" />
                    )}
                </Grid>
            </Container>
        </>
    );
};

export default ParishGeneralPlacementList;
