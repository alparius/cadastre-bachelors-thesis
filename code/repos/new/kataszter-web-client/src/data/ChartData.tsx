interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
