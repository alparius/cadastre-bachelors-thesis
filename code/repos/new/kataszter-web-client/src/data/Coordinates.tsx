interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
