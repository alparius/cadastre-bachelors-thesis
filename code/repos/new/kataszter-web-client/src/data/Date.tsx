interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
