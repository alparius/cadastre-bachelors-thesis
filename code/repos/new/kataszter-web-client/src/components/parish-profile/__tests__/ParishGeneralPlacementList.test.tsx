/*===================
  ParishGeneralPlacementList.unit.test.tsx
 ===================*/
import React from "react";

import IParishPlacement from "../../../data/parish/ParishPlacement";
import { NewTestDate, NewTestParishPlacement } from "../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter } from "../../../util/test/test-wrapper";
import ParishGeneralPlacementList from "../parish-general-data/ParishGeneralPlacementList";

describe("ParishGeneralPlacementList element", () => {
    it("should render a parish placement list", () => {
        const regularPlacements: IParishPlacement[] = [NewTestParishPlacement];
        const container = createComponentWithIntlAndRouter(<ParishGeneralPlacementList placements={regularPlacements} headerKey="test" />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a parish placement list with 'no data' and '-' labels for missing fields", () => {
        const placementWithMissingFields: IParishPlacement = { priestID: 2222, startDate: NewTestDate, endDate: NewTestDate } as any;
        const placementsWithMissingFields: IParishPlacement[] = [placementWithMissingFields];
        const container = createComponentWithIntlAndRouter(
            <ParishGeneralPlacementList placements={placementsWithMissingFields} headerKey="test" />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render an empty list", () => {
        const emptyPlacements: IParishPlacement[] = [];
        const container = createComponentWithIntlAndRouter(<ParishGeneralPlacementList placements={emptyPlacements} headerKey="test" />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
