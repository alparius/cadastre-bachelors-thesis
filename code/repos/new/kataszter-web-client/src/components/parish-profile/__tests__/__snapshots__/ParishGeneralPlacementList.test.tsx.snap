// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParishGeneralPlacementList element should render a parish placement list 1`] = `
Array [
  <h2
    className="ui header"
  >
    <i>
      test
    </i>
  </h2>,
  <div
    className="ui container"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <div
      className="ui grid"
    >
      <div
        className="four column row"
        style={
          Object {
            "marginBottom": "-1em",
          }
        }
      >
        <div
          className="column"
        >
          <b>
            Name
          </b>
        </div>
        <div
          className="column"
        >
          <b>
            Interval
          </b>
        </div>
        <div
          className="column"
        >
          <b>
            Number of people
          </b>
        </div>
      </div>
      <div
        className="four column row"
        style={
          Object {
            "marginBottom": "-1em",
          }
        }
      >
        <div
          className="column"
        >
          <a
            aria-current={null}
            className="hoverLink"
            href="/priests/profile/22222222222"
            onClick={[Function]}
          >
            Janos
          </a>
        </div>
        <div
          className="column"
        >
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
           - 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
        </div>
        <div
          className="column"
        >
          <i>
            12
          </i>
        </div>
        <div
          className="column"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`ParishGeneralPlacementList element should render a parish placement list with 'no data' and '-' labels for missing fields 1`] = `
Array [
  <h2
    className="ui header"
  >
    <i>
      test
    </i>
  </h2>,
  <div
    className="ui container"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <div
      className="ui grid"
    >
      <div
        className="four column row"
        style={
          Object {
            "marginBottom": "-1em",
          }
        }
      >
        <div
          className="column"
        >
          <b>
            Name
          </b>
        </div>
        <div
          className="column"
        >
          <b>
            Interval
          </b>
        </div>
        <div
          className="column"
        >
          <b>
            Number of people
          </b>
        </div>
      </div>
      <div
        className="four column row"
        style={
          Object {
            "marginBottom": "-1em",
          }
        }
      >
        <div
          className="column"
        >
          <i>
            No data
          </i>
        </div>
        <div
          className="column"
        >
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
           - 
          <span
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            2000.06.06
          </span>
        </div>
        <div
          className="column"
        />
        <div
          className="column"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`ParishGeneralPlacementList element should render an empty list 1`] = `
Array [
  <h2
    className="ui header"
  >
    <i>
      test
    </i>
  </h2>,
  <div
    className="ui container"
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <div
      className="ui grid"
    >
      <div
        className="four column row"
        style={
          Object {
            "marginBottom": "-1em",
          }
        }
      >
        <div
          className="column"
        >
          <b>
            Name
          </b>
        </div>
        <div
          className="column"
        >
          <b>
            Interval
          </b>
        </div>
        <div
          className="column"
        >
          <b>
            Number of people
          </b>
        </div>
      </div>
    </div>
  </div>,
]
`;
